### DISCUSSION TITLE
How to export legacy data from Umami.is to Umami for own deployment？

### DISCUSSION THREAD

**Discussion by chairowell** [umami-software/umami#2381] [Q&A] [answered]
I have original data on Umami.is, and now I want to import it into my own Umami deployment. What should I do?

The official website documentation can not find solutions to such problems.

**Comment by mateusfg7** [comment]
+1 on this issue

Looks like the export/import data just works on Umami Cloud.

**Comment by candidexmedia** [comment]
I would like to know as well, and even the other way around (import self-hosted data into umami cloud).

Seems odd that this isn't available in self-hosted deploys.

**Comment by RoversX** [comment, marked as answer]
I wrote a script to finish it, refer to the Youtube video I recorded and the blog post.

https://youtu.be/m3hJ_MJ7Ljo
https://blog.closex.org/posts/29bdb155/
https://github.com/RoversX/umami-csv-import-script
https://huggingface.co/spaces/RoversX/umami_import

**Reply by rsxdalv** [reply]
> I wrote a script to finish it, refer to the Youtube video I recorded and the blog post.
> 
> https://youtu.be/m3hJ_MJ7Ljo https://blog.closex.org/posts/29bdb155/ https://github.com/RoversX/umami-csv-import-script https://huggingface.co/spaces/RoversX/umami_import


```
Remove Primary Key Constraint: This will allow duplicate session_id values in the table. Execute the following command:

ALTER TABLE session DROP CONSTRAINT session_pkey;
Remove Unique Index: Remove the uniqueness constraint on session_id by executing the following command:

DROP INDEX session_session_id_key;
```

Don't do that. Thank you for your script; however, you are not correctly slicing the JOINED data - sessions.csv is not supposed to have duplicates. The easiest fix for that is just to a) during import do nothing on conflicts (PostgreSQL feature) or b) filter out duplicates at python level. As far as I can tell, the earliest date should be the actual 'created_at' value.

**Reply by RoversX** [reply]
Thank you for pointing out that you are right. Sorry I forgot to update the script in the repo I just updated the [script](https://github.com/RoversX/umami-csv-import-script/blob/main/umami_import_no_duplicates.ipynb), there is a version that can generate no duplicate session_id file. and there is another [hugging face space](https://huggingface.co/spaces/ksitarz-sigmoidal/umami_import) that already have that function. Since my original script work, I think, I will just leave that script there.
